Engine/Helper.cs

```diff
@@ -1602,6 +1602,187 @@ public bool GetNamedArgumentAttributeValue(NamedAttributeArgumentAst namedAttrib
             return false;
         }
 
+        /// <summary>
+        /// Gets valid keys of a PowerShell module manifest file for a given PowerShell version
+        /// </summary>
+        /// <param name="powershellVersion">Version parameter with valid values: 5.0, 4.0 and 3.0</param>
+        /// <returns>Returns an enumerator over valid keys</returns>
+        public static IEnumerable<string> GetModuleManifestKeys(Version powershellVersion)
+        {
+            if (powershellVersion == null)
+            {
+                throw new ArgumentNullException("powershellVersion");
+            }
+            if (!IsPowerShellVersionSupported(powershellVersion))
+            {
+                throw new ArgumentException("Invalid PowerShell version. Choose from 3.0, 4.0 or 5.0");
+            }
+            var keys = new List<string>();
+            var keysCommon = new List<string> {
+                    "RootModule",
+                    "ModuleVersion",
+                    "GUID",
+                    "Author",
+                    "CompanyName",
+                    "Copyright",
+                    "Description",
+                    "PowerShellVersion",
+                    "PowerShellHostName",
+                    "PowerShellHostVersion",
+                    "DotNetFrameworkVersion",
+                    "CLRVersion",
+                    "ProcessorArchitecture",
+                    "RequiredModules",
+                    "RequiredAssemblies",
+                    "ScriptsToProcess",
+                    "TypesToProcess",
+                    "FormatsToProcess",
+                    "NestedModules",
+                    "FunctionsToExport",
+                    "CmdletsToExport",
+                    "VariablesToExport",
+                    "AliasesToExport",
+                    "ModuleList",
+                    "FileList",
+                    "PrivateData",
+                    "HelpInfoURI",
+                    "DefaultCommandPrefix"};
+            keys.AddRange(keysCommon);
+            if (powershellVersion.Major == 5)
+            {
+                keys.Add("DscResourcesToExport");
+            }
```

Comment:
please add CompatiblePSEditions for 5.1 or newer.

            if (powershellVersion >= [Version]"5.1")
            {
                keys.Add("CompatiblePSEditions");
            }